examples/adv-generic-train-loop.rs

```diff
@@ -0,0 +1,35 @@
+/// This advanced example shows how to work with dfdx in a generic
+/// training setting.
+use dfdx::prelude::{tensor_collection::TensorCollection, *};
+
+fn classification_train<
+    Inp: Trace<E, D>,
+    Lbl,
+    Model: ModuleMut<Inp::Traced, Error = D::Err> + TensorCollection<E, D>,
+    Opt: Optimizer<Model, D, E>,
+    BatchIter: Iterator<Item = (Inp, Lbl)>,
+    Loss: Backward<E, D, Err = D::Err>,
+    LossFn: FnMut(Model::Output, Lbl) -> Loss,
+    E: Dtype,
+    D: Device<E>,
+>(
```

Comment:
Would love to reduce the number of generics here... they are equal to the number of lines of the actual function sooo that seems excessive